pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/defaultcolumn/BaseDefaultColumnHandler.java

```diff
@@ -388,8 +416,40 @@ protected boolean createColumnV1Indices(String column)
     return true;
   }
 
+  /**
+   * Validates the compatibility of the indexes if the column has the forward index disabled. Throws exceptions due to
+   * compatibility mismatch. The checks performed are:
+   *     - Validate dictionary is enabled.
+   *     - Validate inverted index is enabled.
+   *     - Validate that either no range index exists for column or the range index version is at least 2 and isn't a
+   *       multi-value column (since multi-value defaults to index v1).
+   */
+  protected void validateForwardIndexDisabledConfigsIfPresent(String column, boolean forwardIndexDisabled) {
```

Comment:
Why is this function needed here ? Table config validator should catch the unsupported combinations / mandatory conditions since anyone who wants to toggle the flag and then leverage reload path has to first update the table config successfully on the controller (which should internally use validator)

Reply:
As discussed on the other comment, added this as an additional safeguard. I can remove it though if you think this is unnecessary and the only way is to validate the table configs on the controller. e.g. sometimes we may directly change table config for incompatible schema changes in EI etc. For such cases will the controller catch it for sure?

Reply:
All table config changes must go through controller and it is expected to validate the changed config

Reply:
Ok to do in follow-up PR